### PR TITLE
fix x64 binary url to Microsoft.AzureCLI v 2.52.0

### PR DESCRIPTION
**Pull request by kkamegawa**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] This PR only modifies one (1) manifest
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest .\manifests\m\Microsoft\Azurecli\2.52.0`?
- [ ] Have you tested your manifest locally with `winget install --manifest .\manifests\m\Microsoft\Azurecli\2.52.0`?
- [ ] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/119228)

### CODE REVIEW

File: manifests/m/Microsoft/AzureCLI/2.52.0/Microsoft.AzureCLI.installer.yaml [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 Installers:
-- Architecture: x64
+- Architecture: x86
   InstallerType: msi
   InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.52.0.msi
   InstallerSha256: b410e4fbd8602d46df54c77644209887c9f6bf87df08f062d590cdcd8a96d3b6
+- Architecture: x64
+  InstallerType: msi
+  InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.52.0-x64.msi
+  InstallerSha256: 31a12596bb9f3f5b618c75971b57b2483b6c7c85a813bdd527ca100b6809493c
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/AzureCLI/2.52.0/Microsoft.AzureCLI.installer.yaml:15]:
```suggestion
  InstallerSha256: 31a12596bb9f3f5b618c75971b57b2483b6c7c85a813bdd527ca100b6809493c
  ProductCode: '{C15191B7-6923-4318-AB0F-73AF319257B4}'
```

File: manifests/m/Microsoft/AzureCLI/2.52.0/Microsoft.AzureCLI.installer.yaml [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 
 PackageIdentifier: Microsoft.AzureCLI
 PackageVersion: 2.52.0
+ReleaseDate: 2023-09-05
 Installers:
-- Architecture: x64
+- Architecture: x86
   InstallerType: msi
   InstallerUrl: https://azcliprod.azureedge.net/msi/azure-cli-2.52.0.msi
   InstallerSha256: b410e4fbd8602d46df54c77644209887c9f6bf87df08f062d590cdcd8a96d3b6
```
Comment (mdanish-kh) [inline manifests/m/Microsoft/AzureCLI/2.52.0/Microsoft.AzureCLI.installer.yaml:11]:
```suggestion
  InstallerSha256: b410e4fbd8602d46df54c77644209887c9f6bf87df08f062d590cdcd8a96d3b6
  ProductCode: '{22F9BFD1-9750-4AAE-8E84-6A3F6401AB34}'
```

**Review by mdanish-kh — CHANGES_REQUESTED** [review]
ProductCode helps with package correlation for MSI packages

File: manifests/m/Microsoft/AzureCLI/2.52.0/Microsoft.AzureCLI.installer.yaml (hunk 2 above)
Comment (kkamegawa) [reply manifests/m/Microsoft/AzureCLI/2.52.0/Microsoft.AzureCLI.installer.yaml:11]: @mdanish-kh Thank you! update pull request.

**Review by mdanish-kh — APPROVED** [review]
@kkamegawa Please mark the above conversations as "resolved" for auto-merge process to merge this PR

**Pull request merged**